### ISSUE TITLE
Results blends with code

### ISSUE THREAD

**Issue by vladap** [scalacenter/scastie#133]
Align results to right or place them on the next line or make it optional. Some users probably use Intellij Scala Worksheet which aligns to the right.

**Comment by MasseGuillaume** [comment]
It would be nice to let the user select his instrumentation mode.

* aligned on a panel on the left like scala IDE
* debug mode (current)
* any other relevant mode we can think

**Comment by heathermiller** [comment]
Why would what you call "debug mode" ever be preferable? As far as I know, all other worksheet modes align results to the right. I don't know any that behave like scastie.

**Comment by MasseGuillaume** [comment]
@heathermiller chrome inspector javascript debugger and intellij

![idea](https://d3nmt5vlzunoa1.cloudfront.net/idea/files/2016/10/Screen-Shot-2016-10-27-at-09.32.31.png)

**Comment by heathermiller** [comment]
IntelliJ is on the right side. So is the ScalaIDE for Eclipse. Those are the places where people actually have a concept of a worksheet – I'd not call a debugger a worksheet (I don't think the comparison to the JS debugger is valid)

**Comment by heathermiller** [comment]
Also, just to re-ask my question: Why would what you call "debug mode" ever be preferable?

**Comment by OlegYch** [comment]
i think it's actually preferable to have annotations closer to where they were produced
intellij and eclipse probably output it in separate pane just because it's easier
if the current scastie way turns out to cause too much trouble, and aligned representation is implemented, there at least should be a way to highlight current line (with it's annotation) to provide visual cue

**Comment by heathermiller** [comment]
I disagree that it's desirable to have annotations pushed up against expressions. By panel, I don't mean like an iFrame. I just mean right aligned, not squeezed up against expressions.

So, something like:
![img_4131](https://cloud.githubusercontent.com/assets/687163/23579699/fa6f0384-00f2-11e7-9aad-3e80f37f0db8.JPG)

It doesn't have to be really another pane. Just something to unclutter up individual lines of code.

**Comment by MasseGuillaume** [comment]
what if you add a box around the annotation

**Comment by heathermiller** [comment]
I guess what still makes my eyes itch is that it's so cluttered. Even if there was a box. It's just visually noisy. I like the structured/organized feeling of things where there's a bit of space separating code from result.

If we did the right align thing and had a subtly different color for the line that the cursor was on, there'd be no way to not be able to tell what line the result is on, so that could potentially settle @OlegYch's concerns.

**Comment by vladap** [comment]
I wouldn't mix worksheet and debug. They have different goals. Worksheet displays result for return value of the whole code block. Debug tries to show as much as possible information for each line of code, here it makes sense to have very close.

**Comment by vladap** [comment]
You can take a look on notebooks - Python, Spark like Zeppelin, Databrics, Jupyther. These shows more complex results and shows them on the next line. If results become more complex it can be better choice but I don't think it were is right not.

I would suggest not to get detracted by this much, The problem is that it blends, commonly used solution for Worksheets with simple results is to the right. I would implement to the right. And wait. If users will have demand for left, next line it can be added later with settings. I wouldn't try to guess what users will prefer the most nor implement more complex solution supporting all sorts of bells right now. Let's just solve the problem in the most simple way.

**Comment by vladap** [comment]
If it should ever evolve into something more complex, like notebooks, it would have to be justified by use cases. I would put anything more complex into Scala project and use IDE. As well debug mode would be justified only for more complex use cases.

I think, the goal here is to have online tool with prepared Scala prepared platform with ecosystem core libraries so that it is easy to try them out, learning purposes. I typically use these things to check out how something works in Scala or some library, e.g. to simulate some implicit search scenario. Overtime it could evolve into being able to share working snippets of code in discussions, e.g. like jsfiddle, Allow to inline working code snippets in blogs. Scala doc could use it for examples (inlined).

**Comment by MasseGuillaume** [comment]
This got worse with the UI redesign. I'm bumping the priority.
